assets/shop/js/card_form.js

```diff
@@ -0,0 +1,273 @@
+import * as JSEncrypt from './jsencrypt.min';
+
+const MAX_CARD_NUMBER_LENGTH = 16;
+
+export class CardForm {
+  #form;
+  #cardHolderName;
+  #cardNumber;
+  #cardOperatorIcon;
+  #cardsApi;
+  #cvc;
+  #encryptedCard;
+  #expirationMonth;
+  #expirationYear;
+  #submitButton;
+
+  constructor(selector) {
+    this.#form = document.querySelector(selector);
+    this.#cardHolderName = this.#form.querySelector('[data-tpay-card-holder-name]');
+    this.#cardNumber = this.#form.querySelector('[data-tpay-card-number]');
+    this.#cardOperatorIcon = this.#form.querySelector('[data-tpay-card-operator-icon]');
+    this.#cardsApi = this.#form.querySelector('[data-tpay-cards-api]');
+    this.#cvc = this.#form.querySelector('[data-tpay-cvc]');
+    this.#encryptedCard = this.#form.querySelector('[data-tpay-encrypted-card]');
+    this.#expirationMonth = this.#form.querySelector('[data-tpay-expiration-month]');
+    this.#expirationYear = this.#form.querySelector('[data-tpay-expiration-year]');
+    this.#submitButton = this.#form.querySelector('[type="submit"]');
+    this.#registerEvents();
+    this.#registerFormatters();
+  }
+
+  #registerCvcFormatting() {
+    this.#cvc.addEventListener('keyup', (event) => {
+      const value = event.target.value.replace(/\s/g, '');
+
+      event.target.value = value.slice(0, 3);
+    });
+  }
+
+  #registerCardNumberFormatting() {
+    this.#cardNumber.addEventListener('keypress', (event) => {
+      const value = event.target.value.replace(/\s/g, '');
+
+      if (value.length >= MAX_CARD_NUMBER_LENGTH) {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+    });
+
+    this.#cardNumber.addEventListener('keyup', (event) => {
+      const currentPositionStart = event.target.selectionStart;
+      const currentLength = event.target.value.length;
+
+      const value = event.target.value.replace(/\s/g, '');
+
+      const parts = value.slice(0, 16).match(/\d{1,4}/g) || [];
```

Comment:
```suggestion
      const parts = value.slice(0, MAX_CARD_NUMBER_LENGTH).match(/\d{1,4}/g) || [];
```

let's reuse the const